src/S3/Transfer.php

```diff
@@ -317,15 +317,16 @@ private function uploadMultipart($filename)
 
     private function createS3Key($filename)
     {
-        if (!isset($this->s3Args['Key'])) {
-            return '';
```

Comment:
It is not clear why this method should ever return an empty string - I can't find anything useful in the git history and there doesn't appear to be test coverage. Since this seems to be causing #653 I've assumed it's safe to remove this behaviour - I have not reviewed to see if this has unexpected results elsewhere.